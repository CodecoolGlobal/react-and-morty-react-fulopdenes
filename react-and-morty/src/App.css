html,
body {
  margin: 0;
  /* background-color: rgb(28, 5, 55); */
  background-image: url(https://images.alphacoders.com/876/876589.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-color: rgb(0, 43, 116);
  color: chartreuse;
  overflow-x: hidden;
}

.go-back-button{
  position: absolute;
  top: 5px;
  left: -8.8%;
  margin-top: 10px;
  width: 20%;
  font-size: 18px;
  background-color: rgba(255, 255, 255, 0);
  clip-path: polygon(
      0.5em 0%,
      100% 0,
      100% calc(100% - 0.5em),
      calc(100% - 0.5em) 100%,
      0 100%,
      0% 0.5em);
  font-weight: 700;
  color: black;
  border-color: black;
  mix-blend-mode: overlay;
  cursor: pointer;
  transition: all 500ms;
}

.go-back-button:hover{
  mix-blend-mode: normal;
  color: rgb(40,0,3); 
  border-color: rgb(40,0,3);
  left: 5px;
  transition: all 500ms;
}
