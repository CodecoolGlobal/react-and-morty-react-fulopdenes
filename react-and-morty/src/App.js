import React, { useState } from "react";
import "./App.css";
// import { useCharacters, useLocations } from "./api/useData";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import LandingPage from "./components/Landing-page";
import Characters from "./components/Characters-page";
import Locations from "./components/Locations-page";

function App() {
  const [scrollStatus, setScrollStatus] = useState(window.scrollY);

  window.addEventListener("scroll", () => setScrollStatus(window.scrollY));

  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/">
            <Route index element={<LandingPage />} />
            <Route
              path="characters"
              element={<Characters scrollStatus={scrollStatus} />}
            />
            <Route
              path="locations"
              element={<Locations scrollStatus={scrollStatus} />}
            />
          </Route>
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
