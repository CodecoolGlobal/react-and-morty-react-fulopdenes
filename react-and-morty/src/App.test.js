import React from 'react';
import { getByText, render } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

// test('loads page successfully', () => {
//   render(<App />);
//   const landingImage = screen.getAllByRole("img");
//    expect(landingImage).toHaveLength(1);
// });

// test('more character info shows up on click', () => {

// })
