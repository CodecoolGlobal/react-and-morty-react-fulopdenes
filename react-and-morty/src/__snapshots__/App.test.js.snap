// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THE PAGE LOADS SUCCESSFULLY 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="landing-page"
        >
          <header
            class="landing-page-welcome-text-container"
          >
            <img
              alt="rick and morty logo"
              class="landing-logo-image"
              src="Rick-and-Morty.png"
            />
            <h2>
              Webcatalog
            </h2>
          </header>
          <a
            href="/characters"
            style="text-decoration: none;"
          >
            <div
              class="portal"
            >
              <div
                class="portal-text"
              >
                CHARACTERS
              </div>
              <img
                alt="Character Portal"
                id="char-portal"
                src="portal.gif"
              />
            </div>
          </a>
          <a
            href="/locations"
            style="text-decoration: none;"
          >
            <div
              class="portal"
            >
              <div
                class="portal-text"
              >
                LOCATIONS
              </div>
              <img
                alt="Location Portal"
                id="loc-portal"
                src="portal.gif"
              />
            </div>
          </a>
          <div
            class="page-info"
          >
            <div
              class="mrPB-1-clicked"
            >
              click on me to know more, ooh wee
            </div>
            <img
              alt="mr PoopyButthole"
              class="mrPB-1"
              src="mrPoopyButthole1.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="landing-page"
      >
        <header
          class="landing-page-welcome-text-container"
        >
          <img
            alt="rick and morty logo"
            class="landing-logo-image"
            src="Rick-and-Morty.png"
          />
          <h2>
            Webcatalog
          </h2>
        </header>
        <a
          href="/characters"
          style="text-decoration: none;"
        >
          <div
            class="portal"
          >
            <div
              class="portal-text"
            >
              CHARACTERS
            </div>
            <img
              alt="Character Portal"
              id="char-portal"
              src="portal.gif"
            />
          </div>
        </a>
        <a
          href="/locations"
          style="text-decoration: none;"
        >
          <div
            class="portal"
          >
            <div
              class="portal-text"
            >
              LOCATIONS
            </div>
            <img
              alt="Location Portal"
              id="loc-portal"
              src="portal.gif"
            />
          </div>
        </a>
        <div
          class="page-info"
        >
          <div
            class="mrPB-1-clicked"
          >
            click on me to know more, ooh wee
          </div>
          <img
            alt="mr PoopyButthole"
            class="mrPB-1"
            src="mrPoopyButthole1.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
